// the-router/shim/TheRouterStyle.js
'use strict';

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _theStyle = require("the-style");

/** Style for TheRouter */
var TheRouterStyle = function TheRouterStyle(_ref) {
  var className = _ref.className,
      id = _ref.id,
      options = _ref.options;
  return _react.default.createElement(_theStyle.TheStyle, (0, _extends2.default)({
    id: id
  }, {
    className: (0, _classnames.default)('the-html-style', className),
    styles: TheRouterStyle.data(options)
  }));
};

TheRouterStyle.displayName = 'TheRouterStyle';
TheRouterStyle.propTypes = {
  /** Style options */
  options: _propTypes.default.object
};
TheRouterStyle.defaultProps = {
  options: {}
};

TheRouterStyle.data = function (options) {
  var ThemeValues = _theStyle.TheStyle.ThemeValues;
  var _options$height = options.height,
      height = _options$height === void 0 ? '100%' : _options$height;
  return {
    '.the-router': {
      height: 'inherit'
    },
    '.the-router-hash': {
      height: height
    },
    '.the-router-static': {
      height: height
    }
  };
};

var _default = TheRouterStyle;
exports.default = _default;
