// the-router/shim/TheRouter.js
'use strict';

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _reactRouterDom = require("react-router-dom");

var _theComponentUtil = require("the-component-util");

/**
 * Router of the-components
 */
var TheRouter =
/*#__PURE__*/
function (_React$Component) {
  (0, _inherits2.default)(TheRouter, _React$Component);

  function TheRouter() {
    (0, _classCallCheck2.default)(this, TheRouter);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(TheRouter).apply(this, arguments));
  }

  (0, _createClass2.default)(TheRouter, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var _this = this;

      var _this$props = this.props,
          forceRefresh = _this$props.forceRefresh,
          history = _this$props.history;

      if (history) {
        this.unlisten = history.listen(function () {
          if (forceRefresh) {
            _this.forceUpdate();
          }
        });
      }
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      var _this2 = this;

      var history = this.props.history;

      if (history) {
        var historyChanged = prevProps.history && prevProps.history !== history;

        if (historyChanged) {
          throw new Error('[TheRouter] You cannot change <TheRouter history>');
        }

        this.unlisten = history.listen(function () {
          _this2.forceUpdate();
        });
      }
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      this.unlisten && this.unlisten();
    }
  }, {
    key: "render",
    value: function render() {
      var props = this.props;
      var basename = props.basename,
          children = props.children,
          className = props.className,
          getUserConfirmation = props.getUserConfirmation,
          history = props.history,
          keyLength = props.keyLength;
      var Router = history ? _reactRouterDom.Router : _reactRouterDom.BrowserRouter;
      return _react.default.createElement(Router, {
        basename: basename,
        getUserConfirmation: getUserConfirmation,
        history: history,
        keyLength: keyLength
      }, _react.default.createElement("div", (0, _extends2.default)({}, (0, _theComponentUtil.htmlAttributesFor)(props, {
        except: ['className']
      }), {
        className: (0, _classnames.default)('the-router', className)
      }), children));
    }
  }], [{
    key: "Hash",
    value: function Hash(_ref) {
      var basename = _ref.basename,
          children = _ref.children,
          getUserConfirmation = _ref.getUserConfirmation,
          hashType = _ref.hashType;
      return _react.default.createElement(_reactRouterDom.HashRouter, {
        basename: basename,
        getUserConfirmation: getUserConfirmation,
        hashType: hashType
      }, _react.default.createElement("div", {
        className: "the-router-hash"
      }, children));
    }
  }, {
    key: "Static",
    value: function Static(_ref2) {
      var basename = _ref2.basename,
          children = _ref2.children,
          context = _ref2.context,
          location = _ref2.location;
      return _react.default.createElement(_reactRouterDom.StaticRouter, {
        basename: basename,
        context: context,
        location: location
      }, _react.default.createElement("div", {
        className: "the-router-static"
      }, children));
    }
  }]);
  return TheRouter;
}(_react.default.Component);

TheRouter.propTypes = {};
TheRouter.defaultProps = {};
TheRouter.displayName = 'TheRouter';
var _default = TheRouter;
exports.default = _default;
