// the-router/shim/esm/TheRouterStyle.js
'use strict';

import _extends from "@babel/runtime/helpers/extends";
import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import { TheStyle } from 'the-style';
/** Style for TheRouter */

var TheRouterStyle = function TheRouterStyle(_ref) {
  var className = _ref.className,
      id = _ref.id,
      options = _ref.options;
  return React.createElement(TheStyle, _extends({
    id: id
  }, {
    className: classnames('the-html-style', className),
    styles: TheRouterStyle.data(options)
  }));
};

TheRouterStyle.displayName = 'TheRouterStyle';
TheRouterStyle.propTypes = {
  /** Style options */
  options: PropTypes.object
};
TheRouterStyle.defaultProps = {
  options: {}
};

TheRouterStyle.data = function (options) {
  var ThemeValues = TheStyle.ThemeValues;
  var _options$height = options.height,
      height = _options$height === void 0 ? '100%' : _options$height;
  return {
    '.the-router': {
      height: 'inherit'
    },
    '.the-router-hash': {
      height: height
    },
    '.the-router-static': {
      height: height
    }
  };
};

export default TheRouterStyle;
